scripts/centos-req.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+sudo yum groupinstall -y "Development tools"
+sudo yum install -y gmp-devel mpfr-devel libmpc-devel zlib-devel vim git java java-devel
+curl https://bintray.com/sbt/rpm/rpm | sudo tee /etc/yum.repos.d/bintray-sbt-rpm.repo
+sudo yum install -y sbt texinfo gengetopt
+sudo yum install -y expat-devel libusb1-devel ncurses-devel cmake "perl(ExtUtils::MakeMaker)"
+# deps for poky
+sudo yum install -y python36 patch diffstat texi2html texinfo subversion chrpath git wget
+# deps for qemu
+sudo yum install -y gtk3-devel
+# deps for firemarshal
+sudo yum install -y python36-pip python36-devel rsync
```

Comment:
Not all of the latest dependencies for FireMarshal are listed here.
The authoritative list lives at https://github.com/firesim/FireMarshal/blob/master/centos-requirements.txt (which needs a Debian/Ubuntu equivalent).

@NathanTP: Any preferences on how to manage this?  Some options:
* Remove them and let the user follow the FireMarshal README.
* Inline `centos-dependencies.txt` here.
* Read `centos-dependencies.txt` with a shell redirect (but the FireMarshal submodule is not guaranteed to be initialized).

Reply:
In in favor of inlining (these are recommended system requirements. It shouldn't be a complicated script)

Reply:
I would also prefer inlining, but someone has to remember to synchronize the lists once in a while.

Reply:
synchronize when someone opens an issue that the requirements don't hold anymore :)

Reply:
Historically, we've relied on FireSim to handle this:
https://github.com/firesim/firesim/blob/d801f0ffa7047ef2516ab752f2e2443f322cdaa0/build-setup-nolog.sh#L147

I would prefer not to duplicate this list. It's already enough of a pain to keep everything in sync. If we're going to do this in chipyard, I could put a centos-req.sh in FireMarshal and call that from chipyard and/or firesim to minimize duplicate logic.

As for debian equivalents; I have no interest in writing or maintaining that list. I don't have an ubuntu environment (that I manage) and the setup is more complicated. If someone contributes such a script, I'll happily include it. Even then, we'd need to have an official ubuntu version that we support since various hacks and workarounds will depend on the precise version.

Reply:
I'm not try to do anything robust or perfect. This is a need that came from users (more than one). I believe this PR is better than the current state. If it will get out of sync, we will fix it. I'm not assuming people will relay on this, but if they have a "bare" system, this is better than getting obscure message during the build process. They might still end up getting an obscure build message during the build, but less often.

Reply:
(this script does not get called automatically anywhere. It's just references in the docs)